src/thread/thread_manager.cpp

```diff
@@ -43,11 +43,11 @@
 namespace cubthread
 {
 
-#if defined (NO_GCC_44) || defined (WINDOWS)
```

Comment:
We have had a misunderstanding of NO_GCC_44. We should keep ```#if defined (NO_GCC_44)``` scope. In this case:
```c
thread_local entry *tl_Entry_p = NULL;
```

Same for all blocks of code, except in waiter.wait_for (0) where we don't want to yield.